src/app/(with-main-header)/MobileHeader.tsx

```diff
@@ -22,10 +22,57 @@ const routes = {
 export default function MobileHeader() {
   const pathname = usePathname();
   const currentPathname = pathname.match(/\/\w+/)?.[0] ?? "/";
+  const isLoggedIn = true;
 
   const [isOpen, setIsOpen] = useState(false);
   const [isVisible, setIsVisible] = useState(false);
 
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const buttonRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        buttonRef.current &&
+        !dropdownRef.current.contains(event.target as Node) &&
+        !buttonRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+
+  //임시
+  const [notifications, setNotifications] = useState([
+    {
+      id: 1,
+      nickName: "사용자1",
+      message: "님이 회원님의 게시글에 댓글을 남겼습니다.",
+      isRead: false,
+    },
+    {
+      id: 2,
+      nickName: "사용자2",
+      message: "님이 회원님을 팔로우했습니다.",
+      isRead: true,
+    },
+  ]);
+
+  const toggle = () => setIsOpen(!isOpen);
```

Comment:
토글 함수는 아래처럼 이전 값을 직접 불러와서 사용하는 게 더 안전해요!
```suggestion
  const toggle = () => setIsOpen((prev) => !prev);
```